### ISSUE TITLE
API /tags should provide timestamp for release/tag creation

### ISSUE THREAD

**Issue by Idrinth** [go-gitea/gitea#13004]
- Gitea version (or commit ref): 1.13.0+dev-642-gd8b5235de
- Operating system: linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes https://try.gitea.io/api/v1/repos/idrinth/config-check/tags
  - [ ] No
  - [ ] Not relevant

## Description

Currently the api returns a timestamp for a branch's commit, but no commit timestamp for a tag. This seems inconsistent and makes it unecessarily complicated to find out when something was tagged.

I would expect a structure similar to the following:
```json
[
    {
        "name":"test4issue",
        "id":"fea22d6f670b52cab0685fb531877ecb552d4efb",
        "commit":{
            "url":"https://try.gitea.io/api/v1/repos/Idrinth/config-check/git/commits/fea22d6f670b52cab0685fb531877ecb552d4efb",
            "sha":"fea22d6f670b52cab0685fb531877ecb552d4efb",
			"timestamp": "2020-09-08T18:46:47+02:00"
        },
        "zipball_url":"https://try.gitea.io/Idrinth/config-check/archive/test4issue.zip",
        "tarball_url":"https://try.gitea.io/Idrinth/config-check/archive/test4issue.tar.gz"
    }
]
```

**Comment by zeripath** [comment]
Strangely we already have all the code to do this - but for some reason we don't emit the annotated tags.

I suspect the answer is going to be to that we need to emit a combination of what we currently emit and the github api.

**Comment by 6543** [comment]
I'll have a look at it

**Comment by zeripath** [comment]
```
// ToAnnotatedTag convert git.Tag to api.AnnotatedTag
func ToAnnotatedTag(repo *models.Repository, t *git.Tag, c *git.Commit) *api.AnnotatedTag {
	return &api.AnnotatedTag{
		Tag:          t.Name,
		SHA:          t.ID.String(),
		Object:       ToAnnotatedTagObject(repo, c),
		Message:      t.Message,
		URL:          util.URLJoin(repo.APIURL(), "git/tags", t.ID.String()),
		Tagger:       ToCommitUser(t.Tagger),
		Verification: ToVerification(c),
	}
}
``` 

vs

```
// ToTag convert a git.Tag to an api.Tag
func ToTag(repo *models.Repository, t *git.Tag) *api.Tag {
	return &api.Tag{
		Name:       t.Name,
		ID:         t.ID.String(),
		Commit:     ToCommitMeta(repo, t),
		ZipballURL: util.URLJoin(repo.HTMLURL(), "archive", t.Name+".zip"),
		TarballURL: util.URLJoin(repo.HTMLURL(), "archive", t.Name+".tar.gz"),
	}
}
```

**Comment by lucasqueiroz** [comment]
Hey @6543 hope you don't mind, I opened a Pull Request aiming to fix this.
Please let me know if there's any changes required!
Thanks

**Comment by 6543** [comment]
Not at all 👍

**Comment by lunny** [comment]
Closed by #13026
